chartfx-math/src/main/java/de/gsi/math/MathBaseGenBase.java

```diff
@@ -0,0 +1,464 @@
+                i++;
+        }
+        return i;
+    }
+
+    /**
+     * @param x input
+     * @return Round to nearest integer. Rounds half integers to the nearest
+     */
+    public static int nInt(float x) {
+        int i;
+        if (x >= 0) {
+            i = (int) (x + 0.5f);
+            if ((x + 0.5f == i) && ((i & 1) > 0))
+                i--;
+        } else {
+            i = (int) (x - 0.5);
+            if ((x - 0.5f == i) && ((i & 1) > 0))
+                i++;
+        }
+        return i;
+    }
+
+    public static boolean odd(long a) {
+        return (a & 1) != 0;
+    }
+
+    // Fundamental constants
+    public static final double pi() {
+        return 3.14159265358979323846;
+    }
+
+    public static final double piOver2() {
+        return pi() / 2.0;
+    }
+
+    public static final double piOver4() {
+        return pi() / 4.0;
+    }
+
+    public static double power(double x, double y) {
+        return java.lang.Math.pow(x, y);
+    }
+
+    // Elementary charge
+    public static final double qe() {
+        return 1.602176462e-19;
+    } // [C], [A s]
+
+    public static final double qeUncertainty() {
+        return 0.000000063e-19;
+    }
+
+    // universal gas constant (Na * K)
+    // http://scienceworld.wolfram.com/physics/UniversalGasConstant.html
+    public static final double R() {
+        return k() * nA();
+    } // [J K^-1 mol^-1]
+
+    public static final double radToDeg() {
+        return 180.0 / pi();
+    }
+
+    //// codegen: double -> int, long, short
+    public static double range(double lb, double ub, double x) {
+        return x < lb ? lb : (x > ub ? ub : x);
+    }
+    //// end codegen
+
+    // Dry Air Gas Constant (R / MWair)
+    // http://atmos.nmsu.edu/education_and_outreach/encyclopedia/gas_constant.htm
+    public static final double rgAir() {
+        return (1000.0 * R()) / mwAir();
+    } // [J kg^-1 K^-1]
+
+    public static final double rUncertainty() {
+        return R() * ((kUncertainty() / k()) + (nAUncertainty() / nA()));
+    }
+
+    // Stefan-Boltzmann constant
+    public static final double sigma() {
+        return 5.6704e-8;
+    } // [W m^-2 K^-4]
+
+    public static final double sigmaUncertainty() {
+        return 0.000040e-8;
+    }
+
+    // codegen: double -> float, int, long, short
+    public static double sign(double a, double b) {
+        return (b >= 0) ? abs(a) : -abs(a);
+    }
+    // end codegen
+
+    public static double sin(double arg0) {
+        return java.lang.Math.sin(arg0);
+    }
+
+    public static double sinH(double arg0) {
+        return java.lang.Math.sinh(arg0);
+    }
+
+    public static double sqr(double x) {
+        return x * x;
+    }
+
+    public static double sqrt(double x) {
+        return java.lang.Math.sqrt(x);
+    }
+
+    static final double sqrt2 = 1.4142135623730950488016887242097;
+
+    public static double tan(double arg0) {
+        return java.lang.Math.tan(arg0);
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'java.lang.Math.tan' due to existing implicit import 'java.lang.*'](https://app.codacy.com/manual/GSI/chart-fx/pullRequest?prid=5780327)